controllers/usersignup/usersignup_controller_test.go

```diff
@@ -543,6 +651,65 @@ func TestNonDefaultNSTemplateTier(t *testing.T) {
 		string(metrics.External): 1,
 		string(metrics.Internal): 1,
 	})
+
+	t.Run("second reconcile", func(t *testing.T) {
+		// when
+		res, err = r.Reconcile(context.TODO(), req)
+
+		// then
+		require.NoError(t, err)
+		require.Equal(t, reconcile.Result{}, res)
+
+		// space should be created on the second reconcile
+		spacetest.AssertThatSpace(t, test.HostOperatorNs, "foo", r.Client).
+			Exists().
+			HasSpecTargetCluster("member1").
+			HasTier(customTier.Name)
+
+		t.Run("third reconcile", func(t *testing.T) {
+			// when
+			res, err = r.Reconcile(context.TODO(), req)
+
+			// then
+			require.NoError(t, err)
+			require.Equal(t, reconcile.Result{}, res)
+
+			// spacebinding should be created on the second reconcile
+			spacebindingtest.AssertThatSpaceBinding(t, test.HostOperatorNs, "foo", "foo", r.Client).
+				Exists().
+				HasLabelWithValue(toolchainv1alpha1.SpaceCreatorLabelKey, userSignup.Name).
+				HasLabelWithValue(toolchainv1alpha1.SpaceBindingMasterUserRecordLabelKey, "foo").
+				HasLabelWithValue(toolchainv1alpha1.SpaceBindingSpaceLabelKey, "foo").
+				HasSpec("foo", "foo", "admin")
+
+			// Lookup the userSignup one more and check the conditions are updated
+			err = r.Client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Name, Namespace: req.Namespace}, userSignup)
+			require.NoError(t, err)
+			require.Equal(t, userSignup.Status.CompliantUsername, mur.Name)
+			assert.Equal(t, "approved", userSignup.Labels[toolchainv1alpha1.UserSignupStateLabelKey])
+
+			test.AssertConditionsMatch(t, userSignup.Status.Conditions,
+				toolchainv1alpha1.Condition{
+					Type:   toolchainv1alpha1.UserSignupApproved,
+					Status: v1.ConditionTrue,
+					Reason: "ApprovedAutomatically",
+				},
+				toolchainv1alpha1.Condition{
+					Type:   toolchainv1alpha1.UserSignupComplete,
+					Status: v1.ConditionTrue,
+				},
+				toolchainv1alpha1.Condition{
+					Type:   toolchainv1alpha1.UserSignupUserDeactivatingNotificationCreated,
+					Status: v1.ConditionFalse,
+					Reason: "UserNotInPreDeactivation",
+				},
+				toolchainv1alpha1.Condition{
+					Type:   toolchainv1alpha1.UserSignupUserDeactivatedNotificationCreated,
+					Status: v1.ConditionFalse,
+					Reason: "UserIsActive",
+				})
+		})
```

Comment:
Yeah, I guess the main thing for this test is checking that the customTier is set and since that won't affect the SpaceBinding we can remove it and make the test shorter. Thanks for the suggestion.